Explorer/Assets/DCL/InWorldCamera/CameraReelGallery/CameraReelGalleryController.cs

```diff
@@ -0,0 +1,467 @@
+        {
+            showFailureCts = showFailureCts.SafeRestart();
+            view.errorNotificationView.CanvasGroup.DOKill();
+            view.errorNotificationView.CanvasGroup.alpha = 0f;
+        }
+
+        private async UniTask ShowSuccessNotificationAsync(string message)
+        {
+            HideSuccessNotification();
+            HideFailureNotification();
+
+            view.successNotificationView.SetText(message);
+            view.successNotificationView.Show(showSuccessCts.Token);
+            await UniTask.Delay((int) view.errorSuccessToastDuration * 1000, cancellationToken: showSuccessCts.Token);
+            view.successNotificationView.Hide(false, showSuccessCts.Token);
+        }
+
+        private async UniTask ShowFailureNotificationAsync()
+        {
+            HideSuccessNotification();
+            HideSuccessNotification();
+
+            view.errorNotificationView.Show(showFailureCts.Token);
+            await UniTask.Delay((int) view.errorSuccessToastDuration * 1000, cancellationToken: showFailureCts.Token);
+            view.errorNotificationView.Hide(false, showFailureCts.Token);
+        }
+
+        private MonthGridController GetMonthGrid(DateTime dateTime)
+        {
+            MonthGridController monthGridView;
+            if (monthViews.TryGetValue(dateTime, out MonthGridController monthView))
+                monthGridView = monthView;
+            else
+            {
+                monthGridView = reelGalleryPoolManager.GetGridElement(view.scrollContentRect);
+                monthViews.Add(dateTime, monthGridView);
+            }
+
+            return monthGridView;
+        }
+
+        private async UniTask LoadMorePageAsync(CancellationToken ct)
+        {
+            isLoading = true;
+            Dictionary<DateTime, List<CameraReelResponseCompact>> result = await pagedCameraReelManager.FetchNextPageAsync(ct);
+            float handleHeight = view.verticalScrollbar.handleRect.rect.height;
+
+            foreach (var bucket in result)
+            {
+                MonthGridController monthGridView = GetMonthGrid(bucket.Key);
+
+                List<ReelThumbnailController> thumbnailViews = monthGridView.Setup(bucket.Key, bucket.Value, optionButtonController,
+                    (cameraReelResponse, sprite) => reelThumbnailCache.Add(cameraReelResponse, sprite),
+                    cameraReelResponse => ThumbnailClicked?.Invoke(cameraReelResponse));
+
+                for (int i = 0; i < thumbnailViews.Count; i++)
+                    thumbnailImages[currentSize + i] = thumbnailViews[i];
+
+                currentSize += thumbnailViews.Count;
+            }
+            endVisible = currentSize - 1;
+
+            //Wait for layout to update after the addition of new elements
+            await UniTask.WaitWhile(() => Mathf.Approximately(view.verticalScrollbar.handleRect.rect.height, handleHeight) && currentSize > MINIMUM_ELEMENTS_FOR_WAITING_LAYOUT, cancellationToken: ct);
```

Comment:
Since the layouts needs multiple frames to update themselves (there are many nested ones), the only viable solution I've read about is to recursively call a manual update on them, but I didn't like that either so I went with this since it looked more readable. But I agree that is kind of cheaty